src/goto-programs/remove_function_pointers.cpp

```diff
@@ -355,14 +330,56 @@ void remove_function_pointerst::remove_function_pointer(
     }
   }
 
-  remove_function_pointer(goto_program, function_id, target, functions);
+  ::remove_function_pointer(
+    message_handler,
+    symbol_table,
+    goto_program,
+    function_id,
+    target,
+    functions,
+    add_safety_assertion);
 }
 
-void remove_function_pointerst::remove_function_pointer(
+static std::string function_pointer_assertion_comment(
+  const std::unordered_set<symbol_exprt, irep_hash> &candidates)
+{
+  std::stringstream comment;
+
+  comment << "dereferenced function pointer at must be ";
+
+  if(candidates.size() == 1)
+  {
+    comment << candidates.begin()->get_identifier();
+  }
+  else if(candidates.empty())
+  {
+    comment.str("no candidates for dereferenced function pointer");
```

Comment:
Location here too would be great.

Reply:
My rationale for omitting this is that the location will be provided anyway, I don't think we need to repeat it.